src/storage/secondary/rowset/rowset_iterator.rs

```diff
@@ -95,13 +101,22 @@ impl RowSetIterator {
             dvs,
             column_iterators,
             filter_expr,
+            start_keys: start_keys.to_vec(),
```

Comment:
I think we can simply pass `begin_row_id` and `end_row_id` into RowSetIterator? So that we don't need complex logic to compare chunk against sort key in `next_batch`. `begin_row_id` and `end_row_id` include the range to scan, and we use filter scan to filter data. For example, the outer function determines that block 2 - 3 contains data for user specified sort key. So we can pass block 2's first row_id as begin_row_id, and block 4's first row_id - 1 as end_row_id.

Reply:
May fix in later PRs.